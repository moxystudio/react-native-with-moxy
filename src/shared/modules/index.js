/* istanbul ignore file */

export { default as Button } from './react-native-button';
export { default as SplashScreen } from './react-native-splash-screen';
export { default as SafeAreaHeader } from './react-native-safe-area-header';
export * from './react-theme';
export * from './react-native-locale';
export * from './react-native-app-config';
