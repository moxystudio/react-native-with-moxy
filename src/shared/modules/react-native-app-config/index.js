import AppConfigContext from './context';

const AppConfigConsumer = AppConfigContext.Consumer;

export { default as AppConfigProvider } from './AppConfigProvider';
export { AppConfigConsumer };
export { default as useAppConfig } from './use-app-config';
export { default as withAppConfig } from './with-app-config';
