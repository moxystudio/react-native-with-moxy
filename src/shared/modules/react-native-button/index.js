/* istanbul ignore file */

export { default } from './Button';
