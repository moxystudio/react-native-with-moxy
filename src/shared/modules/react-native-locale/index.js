import LocaleContext from './context';

const LocaleConsumer = LocaleContext.Consumer;

export { default as LocaleProvider } from './LocaleProvider';
export { LocaleConsumer };
export { default as useLocale } from './use-locale';
export { default as withLocale } from './with-locale';
