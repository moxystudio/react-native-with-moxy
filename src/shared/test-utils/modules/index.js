export { AppTree } from './react-app-tree';
