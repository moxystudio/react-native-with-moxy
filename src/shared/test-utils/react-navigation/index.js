/* eslint-env jest */

// Reference documentation: https://reactnavigation.org/docs/navigation-prop/

export const createNavigationProp = () => ({
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: () => ({ remove: jest.fn }),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    setOptions: jest.fn(),
    isFocused: jest.fn(() => false),
});
