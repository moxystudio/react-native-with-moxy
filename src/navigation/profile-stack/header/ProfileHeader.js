import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import { Button, useTheme } from '../../../shared/components';

import buildStyles from './styles';

const ProfileHeader = ({ navigation }) => {
    const { themeStyles } = useTheme();
    const styles = buildStyles(themeStyles);

    const handlePress = useCallback(() => {
        navigation.pop();
    }, [navigation]);

    return (
        <View style={ styles.header }>
            <Button title="Go back" textStyle={ styles.goBack } onPress={ handlePress } />
            <Text style={ styles.title }>Profile Stack</Text>
        </View>
    );
};

ProfileHeader.propTypes = {
    navigation: PropTypes.object.isRequired,
};

export default ProfileHeader;
