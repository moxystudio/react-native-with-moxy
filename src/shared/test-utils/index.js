import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import idObj from 'identity-obj-proxy';
import { AppConfigProvider, ThemeProvider, LocaleProvider } from '../modules';

const localeConfig = {
    locales: [
        {
            id: 'en-US',
            name: 'English',
            messages: idObj,
        },
        {
            id: 'pt-PT',
            name: 'PortuguÃªs',
            messages: idObj,
        },
    ],
    // Consider the following React code:
    //
    // <Text>
    //    <FormattedMessage id="foo" />
    // </Text>
    //
    // In this case, the "foo" key will resolve into the text "bar", and the following will be rendered:
    //
    // <Text>
    //    <Text>
    //       bar
    //    </Text>
    // </Text>
    //
    // This a is necessary workaround because *ByText queries can only look up a string if it's
    // a direct child of a Text component.
    textComponent: Text,
};

const Providers = ({ children }) => (
    <AppConfigProvider>
        <SafeAreaProvider>
            <ThemeProvider>
                <LocaleProvider { ...localeConfig }>
                    { children }
                </LocaleProvider>
            </ThemeProvider>
        </SafeAreaProvider>
    </AppConfigProvider>
);

Providers.propTypes = {
    children: PropTypes.element.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: Providers, ...options });

// Export everything from NTL
export * from '@testing-library/react-native';

// Override render method
export { customRender as render };
