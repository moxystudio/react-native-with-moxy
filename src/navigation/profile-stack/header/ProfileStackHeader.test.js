import React from 'react';
import { mount } from 'enzyme';
import ProfileHeader from './';
import { AppTree } from '../../../shared/test-utils/modules';
import { createNavigationProp } from '../../../shared/test-utils/react-navigation';

const navigation = createNavigationProp();

beforeEach(() => {
    navigation.navigate.mockClear();
});

it('should pop the navigation stack when the back button is pressed', () => {
    const tree = mount(
        <AppTree>
            <ProfileHeader navigation={ navigation } />
        </AppTree>,
    );

    const button = tree.find("[accessibilityLabel='back button']").first();

    button.props().onPress();

    expect(navigation.pop).toHaveBeenCalledTimes(1);
});

it('should have back button', () => {
    const tree = mount(
        <AppTree>
            <ProfileHeader navigation={ navigation } />
        </AppTree>,
    );

    const title = tree.find("[accessibilityLabel='back button']").first().text();

    expect(title).toBe('profile.header.buttons.back');
});

it('should display the screen title', () => {
    const tree = mount(
        <AppTree>
            <ProfileHeader navigation={ navigation } />
        </AppTree>,
    );

    const title = tree.find("[accessibilityLabel='title']").first().text();

    expect(title).toBe('profile.header.title');
});
