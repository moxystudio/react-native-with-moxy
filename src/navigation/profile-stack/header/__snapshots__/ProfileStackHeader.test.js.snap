// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<AppTree>
  <SafeAreaProvider>
    <ThemeContextProvider>
      <ProfileStackHeader
        navigation={
          Object {
            "addListener": [Function],
            "closeDrawer": [MockFunction],
            "dismiss": [MockFunction],
            "dispatch": [MockFunction],
            "getParam": [MockFunction],
            "goBack": [MockFunction],
            "isFocused": [MockFunction],
            "navigate": [MockFunction],
            "openDrawer": [MockFunction],
            "pop": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "popToTop": [MockFunction],
            "push": [MockFunction],
            "replace": [MockFunction],
            "setOptions": [MockFunction],
            "setParams": [MockFunction],
            "state": Object {
              "params": Object {},
            },
            "toggleDrawer": [MockFunction],
          }
        }
      >
        <SafeAreaView
          style={
            Object {
              "height": 100,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 20,
                }
              }
            >
              <Button
                onPress={[Function]}
                textStyle={
                  Object {
                    "color": "#2932D3",
                    "fontFamily": "Lato-Bold",
                    "fontWeight": "normal",
                    "paddingBottom": 6,
                    "paddingTop": 4,
                  }
                }
                title="Go back"
                type="highlight"
              >
                <TouchableHighlight
                  activeOpacity={0.85}
                  delayPressOut={100}
                  onPress={[Function]}
                  underlayColor="black"
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={null}
                    >
                      <Text
                        style={
                          Object {
                            "color": "#2932D3",
                            "fontFamily": "Lato-Bold",
                            "fontWeight": "normal",
                            "paddingBottom": 6,
                            "paddingTop": 4,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#2932D3",
                              "fontFamily": "Lato-Bold",
                              "fontWeight": "normal",
                              "paddingBottom": 6,
                              "paddingTop": 4,
                            }
                          }
                        >
                          Go back
                        </Text>
                      </Text>
                    </View>
                  </View>
                </TouchableHighlight>
              </Button>
              <Text
                style={
                  Object {
                    "color": "#F64F51",
                    "fontFamily": "Lato-Regular",
                    "fontWeight": "normal",
                    "paddingBottom": 6,
                    "paddingTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#F64F51",
                      "fontFamily": "Lato-Regular",
                      "fontWeight": "normal",
                      "paddingBottom": 6,
                      "paddingTop": 4,
                    }
                  }
                >
                  Profile Stack
                </Text>
              </Text>
            </View>
          </View>
        </SafeAreaView>
      </ProfileStackHeader>
    </ThemeContextProvider>
  </SafeAreaProvider>
</AppTree>
`;
